savoirfairelinux/sous-chef
# Improved usability of meal admin

## Fixes #679.

### Changes proposed in this pull request:

* meal/admin.py : inline ingredients under restricted items, sort names alphabetically, list display with group, search in name and search in group to allow filtering.
* meal/models.py : more explicit verbose names for restricted_item and for component.
* delivery/fixtures/sample_data.json : corrected ingredient groups values and restricted_item groups values such that they comply with the allowed choices and therefore will appear in the meal admin.
* meal/locale/fr/LC_MESSAGES/django.po : improved translations.
* meal/tests.py : tests for meal admin search.

### Status


### How to verify this change
1- Stop webserver
2- Backup your current database
3- Start webserver
4- python manage.py flush
5- python manage.py createsuperuser
6- python manage.py loaddata sample_data
7- login to SousChef with admin rights
8- click on Admin in the left menu
9- click on "Components (Dishes and recipes)" and verify that the names are sorted, that the component groups are all displayed and that search works on names and on groups.
10- click on Home then on "Ingredients" and verify that the names are sorted, that the ingredient groups are all displayed and that search works on names and on groups.
11- click on Home then on "Restricted items (Restriction categories)" and verify that the names are sorted, that the restricted_item groups are all displayed and that search works on names and on groups.
12- in the list of restricted_items, click on "green veggies" and verify that all the Incompatibilitys section shows all the ingredients (Arugula, Bok choy, ... zucchini) that correspond to this restriction category.

### Deployment notes and migration

* In the production database, we should check that in table component, column component_group, the values comply with COMPONENT_GROUP_CHOICES = ('main_dish', 'dessert', 'diabetic', 'fruit_salad', 'green_salad', 'pudding',  'compote', 'sides').
* In the production database, we should check that in table ingredient, column ingredient_group, the values comply with INGREDIENT_GROUP_CHOICES = ('meat', 'dairy', 'fish', 'seafood', 'veggies_and_fruits', 'legumineuse', 'grains', 'fresh_herbs', 'spices', 'dry_and_canned_goods', 'oils_and_sauces').
* In the production database, we should check that in table restricted_item, column restricted_item_group, the values comply with RESTRICTED_ITEM_GROUP_CHOICES = ('meat', 'vegetables', 'seafood', 'seeds and nuts', 'other').
